nvme.c

```diff
@@ -221,70 +217,82 @@ void *nvme_realloc(void *p, size_t len)
 	return result;
 }
 
-static void *__nvme_alloc_huge(size_t len, bool *huge)
-{
+#define HUGE_MIN 0x80000
+
+struct nvme_mem_huge {
+	size_t len;
+	bool nvme_malloc; /* p has been allocated using nvme_malloc */
```

Comment:
You could probably drop this field and just recompute it based on `len`?

Reply:
As far I can tell, `len` is not enough to decide if the allocation has used `mmap` or `malloc`. Or do I miss something?

Reply:
Yes, you are correct. I forgot you added the malloc() fallback. There are probably ways you could fit this extra bit in the bits of the other fields that are always 0, but I don't think it's worth the effort.

Reply:
I think a simple solution would be to use `len == 0` to indicate that it needs to be `free()`d, because `free()` doesn't require the length. But don't feel too strongly.